k8s-test-runner/cmd/internal/create.go

```diff
@@ -0,0 +1,161 @@
+	Create.MarkFlagRequired("image-registry-url")
+	Create.Flags().String("test-runner-root-dir", "./", "Test runner root directory with default chart and Dockerfile.testbin")
+}
+
+var (
+	K8sTestRunnerImageName = "k8s-test-runner"
+)
+
+func createRunnerImageRunE(cmd *cobra.Command, args []string) error {
+	testPackage, err := filepath.Abs(args[0])
+	if err != nil {
+		return fmt.Errorf("error getting absolute path: %v", err)
+	}
+	if _, err := os.Stat(testPackage); os.IsNotExist(err) {
+		return fmt.Errorf("folder with the tests does not exist: %s", testPackage)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
+	defer cancel()
+
+	rootDir, err := cmd.Flags().GetString("test-runner-root-dir")
+	if err != nil {
+		return fmt.Errorf("error getting test runner root directory: %v", err)
+	}
+
+	// Get the root directory of the k8s runner
+	rootDirAbs, err := filepath.Abs(rootDir)
+	if err != nil {
+		return fmt.Errorf("error getting absolute path: %v", err)
+	}
+
+	fmt.Printf("Test runner root directory: %s\n", rootDirAbs)
+
+	// Build the test binary
+	testBinPath := filepath.Join(rootDirAbs, "testbin")
+	err = buildTestBinary(ctx, testPackage, testBinPath)
+	if err != nil {
+		return err
+	}
+
+	fmt.Print("Creating docker image for the test binary..\n")
+
+	imageTag, err := cmd.Flags().GetString("image-tag")
+	if err != nil {
+		return fmt.Errorf("error getting test name: %v", err)
+	}
+	buildTestBinDockerImageCmd := exec.CommandContext(ctx, "docker", "build", "--platform", "linux/amd64", "-f", "Dockerfile.testbin", "--build-arg", "TEST_BINARY=testbin", "-t", fmt.Sprintf("%s:%s", K8sTestRunnerImageName, imageTag), ".")
+	buildTestBinDockerImageCmd.Dir = rootDirAbs
+
+	fmt.Printf("Running command from %s: %s\n", rootDirAbs, buildTestBinDockerImageCmd.String())
+
+	buildTestBinDockerImageOutput, err := buildTestBinDockerImageCmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("error building test binary image: %v, output: %s", err, buildTestBinDockerImageOutput)
+	}
+
+	fmt.Printf("Done. Created docker image %s:%s\n", K8sTestRunnerImageName, imageTag)
+
+	registryURL, err := cmd.Flags().GetString("image-registry-url")
+	if err != nil {
+		return fmt.Errorf("error getting ECR registry name: %v", err)
+	}
+
+	err = pushDockerImageToECR(ctx, "us-west-2", registryURL, K8sTestRunnerImageName, imageTag)
+	if err != nil {
+		return fmt.Errorf("error pushing docker image to ECR: %v", err)
+	}
+
+	return nil
+}
+
+func buildTestBinary(ctx context.Context, testPackage, testBinPath string) error {
+	fmt.Printf("Creating test binary for %s\n", testPackage)
+
+	// Create the command to build the test binary
+	buildTestBinCmd := exec.CommandContext(ctx, "go", "test", "-c", "-o", testBinPath)
+	buildTestBinCmd.Dir = testPackage
+
+	// Set environment variables for the build process
+	env := os.Environ()
+	env = append(env, "GOOS=linux", "CGO_ENABLED=0", "GOARCH=amd64")
```

Comment:
Maybe we add flags as params? Not sure every project can be built with `CGO_ENABLED=0`?

Reply:
@skudasov sure, I'll create a separate ticket to do this!